src/Adapter/Category/CategoryDataProvider.php

```diff
@@ -60,25 +61,22 @@ public function __construct(LegacyContext $context)
     /**
      * Get a category.
      *
-     * @param null $idCategory
-     * @param null $idLang
-     * @param null $idShop
+     * @param int|null $idCategory
+     * @param int|null $idLang
+     * @param int|null $idShop
      *
-     * @throws \LogicException If the category id is not set
+     * @throws LogicException If the category id is not set
      *
      * @return Category
      */
     public function getCategory($idCategory = null, $idLang = null, $idShop = null)
     {
         if (!$idCategory) {
-            throw new \LogicException('You need to provide a category id', 5002);
+            throw new LogicException('You need to provide a category id', 5002);
         }
 
         $category = new Category($idCategory, $idLang, $idShop);
-
-        if ($category) {
```

Comment:
I think if the code checks for `$category` it's because sometimes it might be returned as `null`. Not 100% sure as it's ObjectModel.

@kpodemski do you know if it can happen that `$category = new Category($idCategory, $idLang, $idShop);` returns `null` or `false`? if for example Shop ID is bad?

Reply:
`if ($category)` will be always inside as it will be a class instantiation.

Reply:
Category might not exists but this will still return `Category` object, I don't recall any scenario where `new ObjectModel` returns `null`